app/models/timetable.rb

```diff
@@ -0,0 +1,13 @@
+class Timetable < ApplicationRecord
+	belongs_to :library
+
+	def self.batch_update_or_create(library_id, dates, open, close, closed, tbd, note)
+		vals = dates.map!{|day| "('#{library_id}', '#{day}', '#{open}', '#{close}', #{closed}, #{tbd}, '#{note}', NOW(), NOW())"}.join(", ")
```

Comment:
Because this is writing straight sql with parameters provided by the user, we really need to validate the parameters within this method and sanitize the query we create or use interpolation to create the query. Some of these checks are done in the controller, but they should happen again closer to the creation of the sql query in case this method is used in a different controller that doesn't do the same cleanup.